services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +46,46 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  BaseType_t send = xQueueSend(thermalMgrQueueHandle, event, 0); // 0 ticks to wait 
+  if (send != pdPASS) return ERR_CODE_INVALID_QUEUE_MSG/*failed case*/;
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
+void osHandlerLM75BD(void) { 
   /* Implement this function */
+  float* temperature;
+  // Question: a, I using the correct devAddr 
+    error_code_t errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, temperature); // Use your readTempLM75BD function
+
+    if (temperature > HYS_THRESHOLD) {
+        // Temperature is above or equal to the overtemperature threshold
+        overTemperatureDetected();
+    } else if (temperature <= HYS_THRESHOLD) {
+        // Temperature is below or equal to the hysteresis threshold
+        safeOperatingConditions(); 
+    }
```

Comment:
we should not be calling readTemp in an ISR for 2 reasons
1. readTemp will call the i2c helper functions which will block on the I2C bus mutex and we should never be trying to block in an ISR
2. I2C is relatively slow and we want ISRs to always be as short and quick as possible to keep our system responsive